### PR TITLE
ocm: add ManagedCluster list functions

### PR DESCRIPTION
**Pull request by klaskosk**
This commit adds two list functions for the ManagedCluster builder in pkg/ocm. The first is a generic ListManagedClusters to return all ManagedClusters matching the provided ListOptions. The second is a function specific to the o-cloud-manager, which returns a subset of ManagedClusters via its API. This list function returns the ones part of that subset.

Assisted-by: Cursor

## Summary by CodeRabbit

* **New Features**
  * Added managed cluster discovery from ACM hub clusters.
  * Added filtering to identify clusters eligible for O2IMS ingestion.
  * Added validation for cluster availability, required metadata, and valid cluster identifiers.
  * Added special handling for local hub clusters.

* **Tests**
  * Added coverage for cluster listing, eligibility filtering, and edge cases.

### CODE REVIEW

File: pkg/ocm/managedclusterlist_test.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+		}
+
+		builders, err := ListManagedClusters(testSettings, testCase.listOptions...)
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil && len(testCase.listOptions) == 0 {
+			assert.Equal(t, len(testCase.managedClusters), len(builders))
+		}
+	}
+}
```
[the remark this answers is not in the document]
Comment (klaskosk) [reply pkg/ocm/managedclusterlist_test.go:69]: these list tests are in line with what we do for other builders. we'll also switch the package to common-builder shortly which has more robust tests

File: pkg/ocm/managedclusterlist.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+// IsORANEligibleManagedCluster reports whether a ManagedCluster would be ingested by the O2IMS cluster collector.
+func IsORANEligibleManagedCluster(cluster *clusterv1.ManagedCluster) bool {
+	if cluster == nil {
+		return false
+	}
+
+	condition := meta.FindStatusCondition(cluster.Status.Conditions, clusterv1.ManagedClusterConditionAvailable)
+	if condition == nil || condition.Status == metav1.ConditionFalse {
+		return false
+	}
```
[the remark this answers is not in the document]
Comment (klaskosk) [reply pkg/ocm/managedclusterlist.go:96]: we're matching the condition of the oran-o2ims repo here, which just cares that it's not false

**Review by cdvultur — APPROVED** [review]
/lgtm

**Comment by klaskosk** [comment]
thanks @cdvultur!
